README.md

```diff
@@ -14,27 +14,47 @@ Ailly's best feature is rapid prompt engineering iteration. By keeping your prom
+3. Create more numbered files, such as `content/20_knock_knock.md` with "Turn the chicken joke into a knock knock joke."
+4. Run Ailly using NodeJS: `npx @ailly/cli --root content content/20_knock_knock.md`
+  - Ailly will send each file to the configured LLM engine and write its result.
+  - `20_knock_knock.md.ailly.md` will have the new knock knock joke based on thr chickenjome it first wrote!
+
+### System Context
+
+System prompts provide grounding and background information for LLMs.
+There are a number of techniques and "best practices" for developing LLM system prompts.
+In Ailly, these are in files with the name `.aillyrc`, and apply top all files in the current folder when using Ailly.
+These files can also specify properties that control how the LLM prompts are constructed, again for every file.
 
 ### Properties
 
-You can set these properties in a combination of locations, including the command line, .aillyrc, and greymatter. Later settings override earlier settings.
+Ailly generates LLM responses for one file at a time.
+It prepares the conversational history and context by using the file system and folders the file is in.
+There are variations in how Ailly composes the prompt, which can be controlled with several properties.
 
-- **`combined`**: `boolean` If `true`, the file's body is the response and the prompt is in the greymatter key `prompt`. If `false`, the file's body is the prompt and the response is in `{file_name}.ailly.md`. The default is `false`.
+You can set these properties in a combination of locations, including the command line, `.aillyrc` files, and greymatter in each file.
+Later settings override earlier settings.
+
+- **`combined`**: `boolean` If `true`, the file's body is the response and the prompt is in the greymatter key `prompt`. If `false`, the file's body is the prompt and the response is written to `{file_name}.ailly.md`. The default is `false`.
 - **`skip`**: `boolean` If `true`, the prompt is not sent through the LLM (but is part of the context).
 - **`isolated`**: `boolean` If `true`, the LLM inference only includes the system prompt, and not the prior context in this folder.
 - **`context`** `conversation` | `folder` | `none`
-  - `conversation` (default) loads files from the root folder and includes them alphabetically, chatbot history style, before the current file.
-  - `folder` includes all files in the folder at the same level as the current file.
+  - `conversation` (default, unless editing) loads files from the root folder and includes them alphabetically, chatbot history style, before the current file.
+  - `folder` includes all files in the folder at the same level as the current file. (Default when editing.)
   - `none` includes no additional content (including no system context).
+- **`parent`** `root` | `always` | `never`
+  - `root` (default) start the chain of system prompts from here.
+  - `always` include the .aillyrc file in the parent directory as part of this system prompt.
```

Comment:
Is this the immediate parent only? Or are grandparents considered?

Reply:
Immediate parent, which then goes through the same process of default root, can be set to none or parent.